modules/python/python-logmsg.c

```diff
@@ -131,21 +132,15 @@ _py_log_message_ass_subscript(PyObject *o, PyObject *key, PyObject *value)
 
   NVHandle handle = log_msg_get_value_handle(name);
 
-  if (value && is_py_obj_bytes_or_string_type(value))
-    {
-      const gchar *str;
-      py_bytes_or_string_to_string(value, &str);
-      log_msg_set_value(py_msg->msg, handle, str, -1);
-    }
-  else
-    {
-      PyErr_Format(PyExc_ValueError,
-                   "str or unicode object expected as log message values, got type %s (key %s). "
-                   "Earlier syslog-ng accepted any type, implicitly converting it to a string. "
-                   "With this version please convert it explicitly to a string using str()",
-                   value->ob_type->tp_name, name);
-      return -1;
-    }
+  if (!value)
+    return -1;
+
+  gchar *log_msg_value;
+  LogMessageValueType type;
```

Comment:
This is kinda a breaking change as far as I understand.
My understanding that the typing is restricted to 4.0 in the way that everything is set to string before 4.0 flag, but internally even before 4.0 typing is used. But in this case just everything is string, so no change occur. So if at some point type information is injected that yields leaking 4.0 functionality into current version. (This part maybe incorrect, but my comment follows based on this assumption.)

My take on if the 4.0 is not used the python code should still split the error code regarding `str or unicode object` must be injected as value, if the 4.0 is configured it can accept the builtin syslog-ng types.

Reply:
Yes, you are right, this is a breaking change (because we do not use `ValuePairs` here, which would handle the cfg version related conversion for us). However accessing the `GlobalConfig` to check the version at `PyLogMessage` level is really complicated. Maybe we should only merge this to 4.0. I couldn't find a proper solution for this, unfortunately.

Reply:
@MrAnno @bazsi 

As discussed in the meeting, I have fixed this on the accessor side.